catalog/core/catalog-core-sourcemetricsplugin/src/main/java/ddf/catalog/metrics/source/SourceMetricsImpl.java

```diff
@@ -307,7 +310,8 @@ public void addingSource(final Source source, Map props) {
         LOGGER.trace("ENTERING: addingSource");
 
         if (executorPool == null) {
-            executorPool = Executors.newCachedThreadPool();
+            executorPool = new ThreadPoolExecutor(0, Integer.MAX_VALUE, 60L, TimeUnit.SECONDS, new SynchronousQueue<>(),
```

Comment:
❓ Is there a reason you changed this to manual construction of your threadpool instead of just calling Executors.newCachedThreadPool(StandardThreadFactoryBuilder.newThreadFactory("eventProcessorThread"))

Reply:
`newCachedThreadPool` will work here.